app/src/main/java/br/com/nespressohelper/daitan/nespressohelper/BitmapHandler.java

```diff
@@ -0,0 +1,55 @@
+package br.com.nespressohelper.daitan.nespressohelper;
+
+import android.content.res.TypedArray;
+import android.graphics.Bitmap;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.support.v7.app.AppCompatActivity;
+
+import java.io.ByteArrayOutputStream;
+
+/**
+ * Created by estrela on 1/3/18.
+ */
+
+public class BitmapHandler extends AppCompatActivity {
+
+    private Drawable drawable;
+    private Bitmap bitmap;
+    private ByteArrayOutputStream stream;
+    private byte[] bitmapData;
+
+    public byte[] getImageBitmapData(Bitmap bitmap) {
+        if(bitmap == null) {
+            drawable = getDrawable(R.drawable.coffecapsule);
+            bitmap = ((BitmapDrawable) drawable).getBitmap();
+            stream = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+            bitmapData = stream.toByteArray();
+            return bitmapData;
+        }else{
+            stream = new ByteArrayOutputStream();
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+            bitmapData = stream.toByteArray();
+            return bitmapData;
+        }
+    }
+
+    public byte[] getImageBitmapData(int index, int length, TypedArray drawables) {
```

Comment:
Instead of having two methods that are almost the same you could maybe make it more concise. Something like:

```
public static byte[] getImageBitmapData(Context context, @DrawableRes int drawableResId) {
    return getImageBitmapData(ContextCompat.getDrawable(context, drawableResId).getBitmap())
}

public static byte[] getImageBitmapData(@NonNull Bitmap bitmap) {
    ByteArrayOutputStream stream = new ByteArrayOutputStream();
    bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
    return stream.toByteArray();
}
```

And doing the length and null checks on the caller methods maybe. By doing this we avoid mixing the exhibition logic with the image loading logic, separating concerns.